### PR TITLE
(bug fix): Set correct value types for disable check connection check

### LINKED ISSUE #762: (bug): Check connection button always disabled
https://bugzilla.redhat.com/show_bug.cgi?id=1779164 

When adding new fields to the cluster form, this disable function was not updated properly. n

image

### PR DESCRIPTION
**Pull request by ibolton336**
Closes #762

### CODE REVIEW

File: src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx [hunk 1]
```diff
@@ -45,18 +45,38 @@ const valuesHaveUpdate = (values, currentCluster) => {
...
     return true;
   }
-
   const requireSSL = !currentCluster.MigCluster.spec.insecure;
-  const caBundle = currentCluster.MigCluster.spec.caBundle;
   const rawToken = atob(currentCluster.Secret.data.saToken);
   const existingEndpoint = currentCluster.MigCluster.spec.url;
-  const azureResourceGroup = currentCluster.MigCluster.spec.azureResourceGroup;
+
+  const getAzureResourceGroupVal = (azureResourceGroup) => {
```
Comment (eriknelson) [inline src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx]: This looks dangerous. One branch returns a bool, the other returns a string. Should return something of the same type.

File: src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx [hunk 2]
```diff
@@ -45,18 +45,38 @@ const valuesHaveUpdate = (values, currentCluster) => {
...
+    if (azureResourceGroup) {
+      return azureResourceGroup.length > 0
+    } else return '';
```
Comment (eriknelson) [inline src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx]: This is a weird `if...else` format. Can we use something more traditional, with a ternary operator or a standard if/else?

`return azureResourceGroup ? azureResourceGroup.lengh > 0 : '';` (Still need to fix the return type here).

Does the linter like this? We should see if we can update.

Comment (ibolton336) [reply src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx]: +1 updated to ternaries

File: src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx (hunk 1 above)
Comment (ibolton336) [reply src/app/cluster/components/AddEditClusterModal/AddEditClusterForm.tsx]: For isAzure, the controller is expecting a boolean. The azureResourceGroup is a text input. Not sure if i am understanding the issue here?

**Pull request merged**